Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -178,5 +178,257 @@ public void DoGetRequest()
+        // Copy Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Copy.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoCopy)]
+        public SwitchParameter Copy { get; set; }
+
+        // Load Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Load.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoLoad)]
+        public SwitchParameter Load { get; set; }
+
+        // Extract Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Extract.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoExtract)]
+        public SwitchParameter Extract { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            var Client = BigQueryClient.Create(Project);
+
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.DoQuery:
+                    DoQuery(Client);
+                    break;
+                case ParameterSetNames.DoCopy:
+                    DoCopy(Client);
+                    break;
+                case ParameterSetNames.DoLoad:
+                    DoLoad(Client);
+                    break;
+                case ParameterSetNames.DoExtract:
+                    DoExtract(Client);
+                    break;
+                default:
+                    ThrowTerminatingError(new ErrorRecord(
+                        new Exception("You must select a valid BQ Job type."),
+                        "Type Not Found", ErrorCategory.ObjectNotFound, this));
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Query Job main processing function.
+        /// </summary>
+        /// <param name="client">BigQuery client instance to send requests with</param>
+        public void DoQuery(BigQueryClient client)
+        {
+            try
+            {
+                var options = new CreateQueryJobOptions();
+                options.UseLegacySql = UseLegacySql;
+                options.Priority = Priority;
+                options.DestinationTable = DestinationTable;
+                options.DefaultDataset = DefaultDataset;
+                //TODO(ahandley): DryRun option when veneer support is added
+                if (DryRun)
+                {
+                    throw new NotImplementedException(
+                        "DryRun is not yet supported by this cmdlet.  Support will be added soon!");
+                }
+
+                BigQueryJob bqr = client.CreateQueryJob(QueryString, options);
+
+                if (PollUntilComplete)
+                {
+                    bqr.PollUntilCompleted();
+                }
+
+                WriteObject(bqr.Resource);
```

Comment:
These functions should return the Resource to `ProcessRecord`. The `WriteObject` call should be done there.